apis/v1alpha3/tlsroute_types.go

```diff
@@ -86,7 +86,7 @@ type TLSRouteSpec struct {
 	//
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=16
-	Hostnames []Hostname `json:"hostnames,omitempty"`
+	Hostnames []Hostname `json:"hostnames"`
```

Comment:
It's better to not rely on this code generator side effect and add an explicit required marker
```suggestion
	// +required
	Hostnames []Hostname `json:"hostnames,omitempty"`
```
Using `omitempty` on required fields is the most recent API conventions guidance as it prevents mess in the marshalled API and makes the error messages for structured and unstructured clients consistent (field is required for both, instead of field is required for one, and you didn't provide 1 item for the other)

Reply:
Thanks @JoelSpeed ! I've applied and tested `+required`, works as intended!